tests/infra/crypto.py

```diff
@@ -173,3 +173,13 @@ def compute_public_key_der_hash_hex_from_pem(pem: str):
 def compute_cert_der_hash_hex_from_pem(pem: str):
     cert = load_pem_x509_certificate(pem.encode(), default_backend())
     return cert.fingerprint(hashes.SHA256()).hex()
+
+
+def check_key_pair_pem(private: str, public: str, password=None) -> bool:
+    prv = load_pem_private_key(private.encode(), password=password)
+    pub = load_pem_public_key(public.encode())
+    prv_pub_der = prv.public_key().public_bytes(
+        Encoding.DER, PublicFormat.SubjectPublicKeyInfo
+    )
+    pub_der = pub.public_bytes(Encoding.DER, PublicFormat.SubjectPublicKeyInfo)
+    return prv_pub_der == pub_der
```

Comment:
```suggestion
    return prv.public_key() == pub
```

Reply:
That's not right; that comparison is always false because it compares two different key objects (which hold the same key). I initially tried the same and then had to add the complicated conversion to make it work.

Reply:
Really? I could have sworn I used that in the past, hmm..